### DISCUSSION TITLE
Feature to generate index.ts

### DISCUSSION THREAD

**Discussion by iamricard** [stephenh/ts-proto#678] [Ideas]
👋🏼 hi again - i was wondering if y'all had considered a flag to generate an index.ts file per directory generated to be able to do `import * as foo from './path/to/foo'` 👀 where `foo` is a directory that might have multiple files. i understand this would come with gotchas, like clashing message names would be a problem, but there's probably both ways around that (namespacing exports maybe?) or even just erroring out for protos that would present issues.

**Comment by stephenh** [comment]
Hey @iamricard ; sorry for the late reply, but yep that's been asked for a few times:

https://github.com/stephenh/ts-proto/issues/212

I don't think building this would be hard per-se, except for the wrinkle of "how to handle message name conflicts". Which I'm sure there is a reasonable solve there, i.e. detect conflicts and use fully-qualified names or what not, but no one's tackled it yet.

**Reply by iamricard** [reply]
Cool! I have a couple of prototypes with different approaches. I'll see if I can write up what it'd look like and their downsides.

**Reply by Tofandel** [reply]
A very quick and dirty fix is to run `ls -fA protos/ | sed "s/^\(.*\).ts$$/export * from \".\/\1\"/" > ./protos/index.ts

But right now there is a problem because each proto exports shared types (see #1015 )
